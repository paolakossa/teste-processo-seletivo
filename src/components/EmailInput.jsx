import React from 'react';
import { Controller } from 'react-hook-form';
import { Input } from 'antd';
import styled from 'styled-components';
import { CloseOutlined } from '@ant-design/icons';

const InputContainer = styled.div`
  margin-bottom: 1.5rem;
  display: flex;
  flex-direction: column;
`;

const Label = styled.label`
  font-weight: 500;
  margin-bottom: 0.5rem;
`;

const ErrorMessage = styled.span`
  color: #e2001b;
  font-size: 12px;
  margin-top: 4px;
`;

const ErrorIconContainer = styled.span`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  width: 14px;
  height: 14px;
  border-radius: 50%;
  background-color: #e2001b;
  color: #fff;
  font-size: 8px;
`;

const ErrorIcon = () => (
  <ErrorIconContainer>
    <CloseOutlined />
  </ErrorIconContainer>
);

export const EmailInput = ({ control }) => {
  return (
    <InputContainer>
      <Label htmlFor="email">Email</Label>
      <Controller
        name="email"
        control={control}
        defaultValue=""
        rules={{
          required: 'Insira seu email ou usuário',
          pattern: {
            value: /^[a-z0-9._%+-]+@[a-z0-9.-]+\.[a-z]{2,4}$/,
            message: 'Email inválido',
          },
        }}
        render={({ field, fieldState: { error } }) => (
          <>
            <Input
              id="email"
              placeholder="Seu email ou usuário"
              {...field}
              status={error ? 'error' : undefined}
              suffix={error ? <ErrorIcon /> : null}
            />
            {error && <ErrorMessage>{error.message}</ErrorMessage>}
          </>
        )}
      />
    </InputContainer>
  );
};
