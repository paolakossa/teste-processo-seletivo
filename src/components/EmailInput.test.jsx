import React from 'react';
import { render, screen, fireEvent, waitFor } from '@testing-library/react';
import { expect, vi, describe, it } from 'vitest';
import { EmailInput } from './EmailInput';
import { useForm } from 'react-hook-form';

const renderWithForm = () => {
  const Wrapper = () => {
    const { control, handleSubmit, trigger } = useForm();
    return (
      <form onSubmit={handleSubmit(vi.fn())}>
        <EmailInput control={control} />
        <button type="submit">Enviar</button>
        <button type="button" onClick={() => trigger('email')}>
          Validar
        </button>
      </form>
    );
  };

  return render(<Wrapper />);
};

describe('EmailInput Component', () => {
  it('deve renderizar corretamente', () => {
    renderWithForm();
    expect(screen.getByLabelText('Email')).toBeInTheDocument();
    expect(screen.getByPlaceholderText('Seu email ou usuário')).toBeInTheDocument();
  });

  it('deve corresponder ao snapshot', () => {
    const { container } = renderWithForm();
    expect(container).toMatchSnapshot();
  });

  it('deve exibir mensagem de erro ao deixar o campo vazio e enviar o formulário', async () => {
    renderWithForm();

    fireEvent.click(screen.getByText('Enviar'));

    await waitFor(() => {
      expect(screen.getByText('Insira seu email ou usuário')).toBeInTheDocument();
    });
  });

  it('deve exibir erro ao digitar um email inválido', async () => {
    renderWithForm();

    const input = screen.getByPlaceholderText('Seu email ou usuário');
    fireEvent.change(input, { target: { value: 'email-invalido' } });
    fireEvent.click(screen.getByText('Enviar'));

    await waitFor(() => {
      expect(screen.getByText('Email inválido')).toBeInTheDocument();
    });
  });
});
