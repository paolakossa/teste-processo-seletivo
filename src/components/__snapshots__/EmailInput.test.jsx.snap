// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EmailInput Component > deve corresponder ao snapshot 1`] = `
.c0 {
  margin-bottom: 1.5rem;
  display: flex;
  flex-direction: column;
}

.c1 {
  font-weight: 500;
  margin-bottom: 0.5rem;
}

<div>
  <form>
    <div
      class="c0"
    >
      <label
        class="c1"
        for="email"
      >
        Email
      </label>
      <input
        class="ant-input css-dev-only-do-not-override-tjsggz ant-input-outlined"
        id="email"
        name="email"
        placeholder="Seu email ou usuÃ¡rio"
        type="text"
        value=""
      />
    </div>
    <button
      type="submit"
    >
      Enviar
    </button>
    <button
      type="button"
    >
      Validar
    </button>
  </form>
</div>
`;
